File: src/main/java/com/linebot/config/PropertiesConfig.java
package com.linebot.config;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Component
@ConfigurationProperties(prefix = "app")
public class PropertiesConfig {
    private String id;
    private String ownHost;
    private String ownPort;

    // kotlinからgetterが見れないので、一時的にgetterを提供する
    public String getId() { return this.id; }
    public String getOwnHost() { return this.ownHost; }
    public String getOwnPort() { return this.ownPort; }
}

File: src/main/kotlin/com/linebot/config/PropertiesConfig.kt
package com.linebot.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties(prefix = "app")
data class PropertiesConfig(
    var id: String? = null,
    var ownHost: String? = null,
    var ownPort: String? = null,
    var token: String? = null,
    var allowedOriginUri: String? = null
)
